### PR TITLE
First version of the orc8r rock

### PR DESCRIPTION
**Pull request by gruyaume**
## OVERVIEW

Implementation of the orchestrator controller rockcraft built rock. This container image will be used for running the following services in Magma Orchestrator:
- accessd
- analytics
- bootstrapper
- certifier
- configurator
- ctraced
- device
- directoryd
- dispatcher
- eventd
- metricsd
- obsidian
- orchestrator
- service-registry
- state
- streamer
- tenants

A [different](https://github.com/gruyaume/magma-orc8r-lte-rock/pull/1) rock will be created specifically for the lte services:
- ha
- lte
- policydb
- smsd
- subscriberdb
- subscriberdb-cache

## Image size

This image size is 729MB and the upstream image is 1.57GB which is around 50% reduction in size.

## Base

This image is built on Ubuntu 22.04 and the upstream image is built on Ubuntu 16.04.

### CODE REVIEW

File: .github/workflows/push_main.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    branches:
+      - main
+
+jobs:
+  build:
+    uses: ./.github/workflows/build.yaml
+
+  publish:
+    needs: build
+    uses: ./.github/workflows/publish.yaml
```
Comment (ghislainbourgeois) [inline .github/workflows/push_main.yaml]: We are not testing before publishing, things could have changed between the CI run on the PR and the CI on main.

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+      - name: Install tox
+        run: |
+          pip install tox
+
+      - name: Tests
+        run: |
+          image_name="$(yq '.name' rockcraft.yaml)"
+          version="$(yq '.version' rockcraft.yaml)"
+          docker run -d -p 8080:10112 ghcr.io/gruyaume/${image_name}:${version}
```
Comment (ghislainbourgeois) [inline .github/workflows/test.yaml]:
```suggestion
          docker run -d -p 8080:10112 ghcr.io/canonical/${image_name}:${version}
```

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      - name: Import
+        run: |
+          image_name="$(yq '.name' rockcraft.yaml)"
+          version="$(yq '.version' rockcraft.yaml)"
+          rock_file=$(ls *.rock | tail -n 1)
+          sudo skopeo \
+            --insecure-policy \
+            copy \
+            oci-archive:"${rock_file}" \
+            docker-daemon:"ghcr.io/gruyaume/${image_name}:${version}"
```
Comment (ghislainbourgeois) [inline .github/workflows/test.yaml]:
```suggestion
            docker-daemon:"ghcr.io/canonical/${image_name}:${version}"
```

**Review by ghislainbourgeois — CHANGES_REQUESTED** [review]

File: .github/workflows/push_main.yaml (hunk 1 above)
Comment (gruyaume) [reply .github/workflows/push_main.yaml]: Fixed

**Review by ghislainbourgeois — APPROVED** [review]

**Pull request merged**